src/screens/swap/swap-form/swap-form.tsx

```diff
@@ -167,18 +170,43 @@ export const SwapForm: FC<SwapFormProps> = ({ inputToken }) => {
     }
   }, [searchValue, assetsList]);
 
+  const minimumReceivedAmount = useMemo(() => {
+    if (bestTradeWithSlippageTolerance.length > 0) {
+      const lastTradeOperation = bestTradeWithSlippageTolerance[bestTradeWithSlippageTolerance.length - 1];
+
+      return mutezToTz(lastTradeOperation.bTokenAmount, outputAssets.asset.decimals);
+    }
+
+    return undefined;
+  }, [bestTradeWithSlippageTolerance, outputAssets.asset.decimals]);
+
+  useEffect(() => {
+    if (isDefined(bestTrade)) {
+      const bestTradeOutput = getTradeOutputAmount(bestTrade);
```

Comment:
move this all logic to getTradeOutputAmount()

Reply:
can't modify this fn due to different use-cases in exchange rate calculation and output calculation